spec.md

```diff
@@ -0,0 +1,251 @@
+
+- **`org|company`** - represents an open source foundation (`oci`, `cncf`) or a company (`microsoft`, `ibm`).
+- **`objectType`** - a value representing the short name of the type. The combination of the org|company and object type should provide human identification and uniqueness for computers indexing.
+- **`optionalSubType`** - provides additional extensibility of an `objectType`
+- **`version`** - provides artifact authors the ability to revision their schemas and formatting, enabling tools to identify which format they will process.
+
+## Defining Supported Layer Types
+
+Artifacts are intended to have content. The content of an artifact is represented through a collection of blobs as provided by the [distribution spec][distribution-spec]. How the blobs are re-constituted, and whether the blobs are ordinal layers is a decision of the artifact author.
+
+As an example, [OCI Images][image-layer] are represented through an ordinal collection of tar archives. Each blob represents a layer. Each layer overlays the previous layer. These blobs may be further compressed.
+
+**Note**: *For the purposes of this spec, the terms **layer** and **blob** are interchangable.*
+
+Other artifacts may be represented as a single file, such as config artifact representing a deployment specification. Other artifacts may include a config blob, and a collection of binaries compressed as another blob. By separating the blob, the artifact author can benefit from layer de-duplication and concurrent downloading of each blob.
+
+### Layer File Format
+
+Layers are persisted as blobs within registries. The blobs can be a single file or a collection of files (`.tar`). The blob format is up to the artifact author. They may choose to persist individual files with standard or custom formats, such as `.config`, `.json`, `.bin`, `.sif`, or store a collection of files as `.tar` archive. Optionally, the author MAY choose to compress the blob. The extension of the `layer.mediaType` MUST relfect the format of the blob.
+
+### Layer Versioning
+
+Layers MUST define a version to future proof new enhancements that may evolve. How the artifact tooling processes forward and backward compatibility is up to the artifact and tooling authors.
+
+### Defining Layer Types
+
+Artifact **layers types** are not expected, nor required to be shared across different **artifact types**. A layer type that represents an OCI Image is not expected to be shared with a Helm Chart. However, two helm charts that have the same content MAY be de-duped within the registry. The **layer type** is the defintion of the contents. To differentiate layers between artifact types, artifact authors SHOULD define unique artifact layer types.
+
+Artifact layer types follow the same convention as manifests, utilizing the `layer.mediaType` with the following format:  
+`application/vnd.`[org|company]`.`[layerType]`.`[layerSubType]`.layer.`[version]`.`[fileFormat]`+`[OPTIONAL compressionFormat]
```

Comment:
hmm.. Should this be generalized to a modifier suffix?  i.e. for encryption usecase, and future extensibility.

Reply:
I'd rather use `.blob.` instead of ~`.layer.`~ in the , since "layer" is normal in the container image context, but completely misleading in a generic artifact context.

Reply:
@Silvanoc, that's a great suggestion. Layer has specific meaning to container images that overlay the previous layers. In the generic artifact type, these aren't overlays. Or, rather, if they are, they should use .layer. I'll update the spec to say, use `.layer` when the layers are overlays, similar to oci-image, and use `.blob` for independent references.

Reply:
Hi @Silvanoc, I wrote a version of this with blob as the reference, but there's too many instances where layers are referenced in the spec. Hopefully, I captured the layers aren't necessarily ordinal, based on the artifact type, and they are persisted as blobs. I just couldn't find a way to refer to blobs and then `layer.mediaType` as defined by the [image spec](https://github.com/opencontainers/image-spec/blob/master/manifest.md#image-manifest-property-descriptions)
> layers array of objects
> Each item in the array MUST be a descriptor. The array MUST have the base layer at index 0. Subsequent layers MUST then follow in stack order (i.e. from layers[0] to layers[len(layers)-1]). The final filesystem layout MUST match the result of applying the layers to an empty directory. The ownership, mode, and other attributes of the initial empty directory are unspecified.